### ISSUE TITLE
Improve song library sort

### ISSUE THREAD

**Issue by dudeofawesome** [StarGazer1258/BeatDrop#26]
It looks like songs are sorted by their ID in the BeatDrop library, which is pretty confusing as a user. I think it would be a huge improvement if songs could be sorted by name and/or artist name.

I'd be happy to implement this myself if I could be pointed in the right direction.

**Comment by Neededchicken38** [comment]
I agree, I would love to have a sorting feature, sort by PP, Most Popular, difficulty. Or even a feature that allows users to subscribe to mappers, therefore all maps from that mapper would be downloaded. Beat Drop is amazing. Thank you BB

**Comment by nvirth** [comment]
+1

**Comment by Rohaq** [comment]
+1 for sorting, but filters would also be a welcome addition: It'd be nice to be able to filter by artist, difficulty, genre (if that's in the metadata) etc.

**Comment by zxsdklfhskyrvk** [comment]
It would be great to be able to filter and sort on various song aspects.  Ideally you should be able to filter on multiple items/fields.

Song Difficulty supported (you choose one or more of the standard difficulties)
Current download count of song (on bsaber)
Current upvote count of song (on bsaber)
Age of song (date since placed on bsaber) - maybe choices for last X days/weeks/months
Song Artist
Song Mapper
Song Genre (from bsaber?)

**Comment by StarGazer1258** [comment]
This has stalled because there is little that can be done. True sorting/filtering on BeatSaver song cannot be done locally, it must be done by the server. Unless BeatSaver implements these functions, there is nothing to be done unfortunately.

**Comment by zxsdklfhskyrvk** [comment]
I am referring to the personal Library,  not the Beatsaver listing.  I cannot imagine why that would require anything server-side.  
You could parse all the local json/dat (during library build ideally) and build tables of the various values of interest.  

Hypothetically, you could collect the "tag" (or require it in the metadata file) to be able to reach out and grab additional song info from bsaber/beatsaver online.  Not sure if they have an API for that or not, TBH.
